### ISSUE TITLE
False positive on only_used_in_recursion

### ISSUE THREAD

**Issue by DidiBear** [rust-lang/rust-clippy#9384]
### Summary

The lint `only_used_in_recursion` is raising an error but I cannot figure out why.
You can check [this Rust Playground here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=84dc61570c57dc01d0478af6d7698154).

### Lint Name

only_used_in_recursion

### Reproducer

I tried this code ([Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=84dc61570c57dc01d0478af6d7698154)):

```rust
#![deny(clippy::only_used_in_recursion)]

struct Value(i32);

fn set_to_last(mut value: Value, values: &[i32]) {
    value.0 = *values.iter().last().unwrap(); 
}

```

I saw this happen:

```
error: parameter is only used in recursion
 --> src/lib.rs:6:16
  |
6 | fn set_to_last(mut value: Value, values: &[i32]) {
  |                ^^^^^^^^^ help: if this is intentional, prefix with an underscore: `_value`
  |
note: the lint level is defined here
 --> src/lib.rs:2:9
  |
2 | #![deny(clippy::only_used_in_recursion)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion
```

I expected no error.

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-apple-darwin
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by Jarcho** [comment]
This should be fixed in #8804

**Comment by kraktus** [comment]
[rust playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cf88df1a2dd8db7206e5d7c78c9556f2) This is indeed fixed on the latest nightly clippy, can be closed 👍
